units/XNB2302/XNB2302_script.lua

```diff
@@ -1,32 +1,42 @@
+-- T4 artillery
+--this one is supposed to be N...FactoryUnit to use buildeffects, but for some reason it crashes game, will make buildable effects later, for now i need unit to work
 local NStructureUnit = import('/lua/nomadsunits.lua').NStructureUnit
+local NUtils = import('/lua/nomadsutils.lua')
 
---- Tech 3 Stationary Artillery
----@class XNB2302 : NStructureUnit
 XNB2302 = Class(NStructureUnit) {
-
-    ---@param self XNB2302
+    
     OnCreate = function(self)
         NStructureUnit.OnCreate(self)
     end,
-
-    ---@param self XNB2302
+    
     OnDestroy = function(self)
         NStructureUnit.OnDestroy(self)
     end,
 
-    ---@param self XNB2302
-    ---@param instigator Unit
-    ---@param damageType DamageType
-    ---@param overkillRatio number
-    OnKilled = function(self, instigator, damageType, overkillRatio)
-        NStructureUnit.OnKilled(self, instigator, damageType, overkillRatio)
+    OnKilled = function(self, instigator, type, overkillRatio)
+        NStructureUnit.OnKilled(self, instigator, type, overkillRatio)
+		
+		--Destroy all orbital on Countdown's death
+		local units = self:GetAIBrain():GetListOfUnits(categories.EXPERIMENTAL, false, false)
+		local toRemove = {}
+		for i, unit in ipairs(units) do
+			local name = unit:GetBlueprint().General.UnitName
+			
+			--abort if another Countdown still exists
+			if unit:GetUnitId() == "xnb2302" and unit:GetEntityId() ~= self:GetEntityId() then return end
+			if unit:GetUnitId() == "xno2302" then
+				toRemove[i] = unit
+			end
+		end
+		
+		for _, unit in pairs(toRemove) do
+			unit:Kill()
+		end
```

Comment:
- You can use the unit's blueprint id (which is lowercase) as a category
- FAF adds `Unit.Brain` as a quicker replacement for `Unit:GetAIBrain()`
- FAF adds `Unit.Dead` to check death status. For this station, it will be set to true in the `OnKilled` you called earlier. For other stations, we have to check it just in case multiple stations die at the same time and a different station exists as a unit but is in its death animation (the entity ID check doesn't work for that case).
```suggestion
                -- If no other Countdowns exist, kill all Deadline satellites
                local brain = self.Brain
		local stations = brain:GetListOfUnits(categories.xnb2302, false, true)
		for _, station in stations do
		  if not station.Dead then
		    return
		  end
		end
		
                local satellites = brain:GetListOfUnits(categories.xno2302, false, true)
	        for _, satellite in satellites do
		  satellite:Kill()
		end
```

Reply:
The satellites cost a lot, so maybe it can be more fun to disable their weapons, movement, and selectability once all stations are dead. You would also have to handle re-enabling all satellites once a station is rebuilt.
```lua
satellite:SetAllWeaponsEnabled(false)
satellite:SetUnSelectable(true)
satellite:SetImmobile(true)
```